src/OpenTelemetry.AutoInstrumentation/Loading/MySqlDataInitializer.cs

```diff
@@ -17,22 +17,27 @@
 #if NETCOREAPP3_1_OR_GREATER
 
 using System;
+using OpenTelemetry.AutoInstrumentation.Plugins;
 
 namespace OpenTelemetry.AutoInstrumentation.Loading;
 
 internal class MySqlDataInitializer : InstrumentationInitializer
 {
-    public MySqlDataInitializer()
+    private readonly PluginManager _pluginManager;
+
+    public MySqlDataInitializer(PluginManager pluginManager)
         : base("MySql.Data")
     {
+        _pluginManager = pluginManager;
     }
 
     public override void Initialize(ILifespanManager lifespanManager)
     {
         var instrumentationType = Type.GetType("OpenTelemetry.Instrumentation.MySqlData.MySqlDataInstrumentation, OpenTelemetry.Instrumentation.MySqlData");
-        var optionsInstrumentationType = Type.GetType("OpenTelemetry.Instrumentation.MySqlData.MySqlDataInstrumentationOptions, OpenTelemetry.Instrumentation.MySqlData");
 
-        var options = Activator.CreateInstance(optionsInstrumentationType);
+        var options = new OpenTelemetry.Instrumentation.MySqlData.MySqlDataInstrumentationOptions();
+        _pluginManager.ConfigureOptions(options);
```

Comment:
This could crash the application at runtime if the plugin throws an exception. Right?